lib/semant.ml

```diff
@@ -44,17 +44,18 @@ let check (binds, funcs, stmts): sprogram =
   in
 
   let check_func fn = 
-    let scopes = ref [globals] in
+    let scopes = ref [Hashtbl.create 10; globals] in
```

Comment:
here was the bug, I forgot to create a scope for the function so in the "base" scope for the function it was actually adding to the global scope...

Reply:
oh lolllll good fix